native/src/main/java/io/ballerina/lib/avro/Avro.java

```diff
@@ -0,0 +1,98 @@
+import io.ballerina.runtime.api.utils.ValueUtils;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.api.values.BTypedesc;
+import org.apache.avro.Schema;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static io.ballerina.lib.avro.Utils.AVRO_SCHEMA;
+import static io.ballerina.lib.avro.Utils.DESERIALIZATION_ERROR;
+import static io.ballerina.lib.avro.Utils.JSON_PROCESSING_ERROR;
+
+public final class Avro {
+
+    private Avro() {}
+
+    public static void generateSchema(BObject schemaObject, BString schema) {
+        Schema.Parser parser = new Schema.Parser();
+        Schema nativeSchema = parser.parse(schema.getValue());
+        schemaObject.addNativeData(AVRO_SCHEMA, nativeSchema);
+    }
+
+    public static Object toAvro(BObject schemaObject, Object data) {
+        Schema schema = (Schema) schemaObject.getNativeData(AVRO_SCHEMA);
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            Object jsonObject = generateJsonObject(data, schema, objectMapper);
+            if (jsonObject == null) {
+                return ValueCreator.createArrayValue(new byte[]{0, 0});
+            } else if (Objects.equals(schema.getType(), Schema.Type.FIXED)) {
+                return ValueCreator.createArrayValue(((BArray) jsonObject).getByteArray());
+            }
+            byte[] avroBytes = (new AvroMapper()).writer(new AvroSchema(schema)).writeValueAsBytes(jsonObject);
+            return ValueCreator.createArrayValue(avroBytes);
+        } catch (JsonProcessingException e) {
+            return Utils.createError(JSON_PROCESSING_ERROR, e);
+        }
+    }
+
+    public static Object fromAvro(BObject schemaObject, BArray payload, BTypedesc typeParam) {
+        Schema schema = (Schema) schemaObject.getNativeData(AVRO_SCHEMA);
+        byte[] avroBytes = payload.getByteArray();
+        if (Objects.equals(schema.getType(), Schema.Type.FIXED)) {
+            return ValueUtils.convert(ValueCreator.createArrayValue(avroBytes), typeParam.getDescribingType());
+        }
+        JsonNode deserializedJsonString;
+        try {
+            AvroMapper mapper = new AvroMapper();
+            deserializedJsonString = mapper.readerFor(Object.class).with(new AvroSchema(schema)).readTree(avroBytes);
+        } catch (IOException e) {
+            return Utils.createError(DESERIALIZATION_ERROR, e);
+        }
+        Object jsonObject = JsonUtils.parse(deserializedJsonString.toPrettyString());
+        return ValueUtils.convert(jsonObject, typeParam.getDescribingType());
+    }
+
+    private static Object generateJsonObject(Object data, Schema schema,
+                                             ObjectMapper objectMapper) throws JsonProcessingException {
+        if (Objects.equals(schema.getType(), Schema.Type.NULL) || Objects.equals(schema.getType(), Schema.Type.FIXED)) {
```

Comment:
```suggestion
        if (Schema.Type.NULL.equals(schema.getType()) || Schema.Type.FIXED.equals(schema.getType()) {
```

check for other places as well.